pkg/apis/sources/v1alpha1/awscodecommit_lifecycle.go

```diff
@@ -37,7 +39,7 @@ func (s *AWSCodeCommitSource) GetSink() *duckv1.Destination {
 }
 
 // GetARN implements AWSEventSource.
-func (s *AWSCodeCommitSource) GetARN() string {
+func (s *AWSCodeCommitSource) GetARN() apis.ARN {
 	return s.Spec.ARN
 }
```

Comment:
`GetARN()` returns the typed ARN instead of a string, so we can now remove all calls to `arn.Parse()` from the reconciliation code.